build-tools/xaprepare/xaprepare/Steps/Step_InstallDotNetPreview.cs

```diff
@@ -20,15 +20,6 @@ protected override async Task<bool> Execute (Context context)
 			var dotnetPreviewVersion = context.Properties.GetRequiredValue (KnownProperties.DotNetPreviewVersionFull);
 			var dotnetTestRuntimeVersion = Configurables.Defaults.DotNetTestRuntimeVersion;
 
-			// Delete any custom Microsoft.Android packs that may have been installed by test runs. Other ref/runtime packs will be ignored.
-			var packsPath = Path.Combine (dotnetPath, "packs");
-			if (Directory.Exists (packsPath)) {
-				foreach (var packToRemove in Directory.EnumerateDirectories (packsPath).Where (p => new DirectoryInfo (p).Name.Contains ("Android"))) {
-					Log.StatusLine ($"Removing Android pack: {packToRemove}");
-					Utilities.DeleteDirectory (packToRemove);
-				}
-			}
-
```

Comment:
Does it say somewhere why this was removed?

Reply:
Indeed, why is this code going away?

Reply:
Sorry I didn't explain this in the description -- it has to do with test environment setup ordering for the APK test job. The `setup-test-environment.yaml` template installs our packs into the dotnet 5 test location, however this subsequent `Required` scenario run will delete them:
https://github.com/xamarin/xamarin-android/blob/master/build-tools/automation/azure-pipelines.yaml#L399

As an alternative to removing this snippet, we could either:
1) Extract the packs again after running the Required scenario required for APK tests
2) Build and run xaprepare -s:Required before test environment setup

Maybe 2 is better here?

Reply:
I think as long as each CI stage deletes and create a fresh `~/android-toolchain/dotnet` directory, it will be fine.

I think we've seen a couple cases where leaving old `dotnet`'s around causes problems, such as:

https://github.com/dotnet/runtime/issues/42039

Reply:
WRT 42039 the SDK clean up logic will remain unchanged: https://github.com/xamarin/xamarin-android/blob/8b367fdba4507be34c074611c2fcd618b5ab4815/build-tools/xaprepare/xaprepare/Steps/Step_InstallDotNetPreview.cs#L26

I'll rework this a bit to keep the Android pack clean up logic untouched as well.

Reply:
The required scenario is only used by this APK tests job, so I removed the .NET Preview installation step from this scenario. The .NET preview will still be installed as part of the test environment setup yaml template.